src/main/scala/com/fulcrumgenomics/umi/UmiConsensusCaller.scala

```diff
@@ -245,11 +245,36 @@ trait UmiConsensusCaller[ConsensusRead <: SimpleRead] {
     // end of the read, or the end of the insert, whichever is shorter!
     val len = {
       var index = if (!rec.isFrPair) trimToLength - 1 else {
-        // Adjust the insert size based on clipping on the 5' end of the given read.
-        // This *does not* consider any clipping on the 5' end of its mate.
-        val fivePrimeClipping  = cigar.takeWhile(_.operator.isClipping).filter(_.operator == CigarOperator.S).sumBy(_.length)
-        val adjustedInsertSize = abs(rec.insertSize) + fivePrimeClipping
-        min(adjustedInsertSize, trimToLength) - 1
+        // If this read reads through the template (i.e. past the start of the mate), then we need to trim bases on the
+        // 3' end.  If not, then we can use the full read. This *does not* consider any clipping on the 5' end of its mate.
+        val adjustedReadLength = {
+          if (rec.positiveStrand) {
+            val mateEnd = rec.mateEnd.getOrElse(rec.start + abs(rec.insertSize) - 1)
+            if (rec.end >= mateEnd) {
+              // the # of bases to keep is the last read base of where the mate ends
+              rec.readPosAtRefPos(pos=mateEnd, returnLastBaseIfDeleted=false)
+            } else {
+              // Find where the end of the read goes past the mate start, if at all
+              val refPosAtReadEnd  = min(mateEnd, rec.end) // last 3' base
+              rec.readPosAtRefPos(pos=refPosAtReadEnd, returnLastBaseIfDeleted=false)
```

Comment:
It's a while since I've been through this code, but this will remove any trailing clipping, insertions, and maybe deletions too.  Is that intentional?  I.e. if the cigar is `80M20S`, say because the read runs off into a large insertion, this will return `80`, and cause the read to get trimmed to 80 bases, no?